opendevin/events/action/commands.py

```diff
@@ -40,6 +40,15 @@ def message(self) -> str:
     def __str__(self) -> str:
         return f'**CmdKillAction**\n{self.command_id}'
 
+    def __eq__(self, other: object) -> bool:
+        """
+        Compare two CmdKillAction objects ignoring the command_id/pid.
+        """
+        if isinstance(other, CmdKillAction):
+            # for loop detection purpose, we care about running the same command, not the same pid
+            return self.thought == other.thought
```

Comment:
I am just imagining how people could leverage this characteristic to write bugs 💭 

Say LLM returns a list of PIDs to kill. If one constructs CmdKillActions out of those PIDs and adds them to a set, then a bug might happen?

Reply:
The alternative argument is: CmdKillAction has 11 lines of code, less than 20 with eq. The docstring is one line. Methods should be documented automatically on GitHub.io afaik, so one can find eq there too. If one didn't read any of the three... is it fair that they add the same element to a set? Not to mention test that their set does the job.

I don't know if I agree with what I just said. I actually do believe that matching implicit developer expectations is generally better... If you feel the same, I can reimplement this just for stuck.

Reply:
Thinking about it a bit more, let's not do this! Dataclass eq behavior is well known and well documented. And, one doesn't call this explicitly, client code just uses `==`, and won't see any custom docstring or whatever.

I'll update the PR in a bit. Thank you!

Reply:
Fixed.

Reply:
I wouldn't be surprised people would write bugs without looking at doc :P 

The problem is there's no way to prevent people from doing that